platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsAccessFromGroovyDslIntegrationTest.groovy

```diff
@@ -689,6 +689,9 @@ class IsolatedProjectsAccessFromGroovyDslIntegrationTest extends AbstractIsolate
         """
 
         when:
+        if (expr == "dependencies.project([path: ':', configuration: 'test'])") {
+            executer.expectDocumentedDeprecationWarning("Accessing the build dependencies of a project dependency has been deprecated. This will fail with an error in Gradle 9.0. Add the dependency to a resolvable configuration and use the configuration to track task dependencies. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#deprecate_self_resolving_dependency")
```

Comment:
What's accessing the build dependencies here?

Reply:
Thats the purpose of this test. It calls `<whatever>.buildDependencies.getDependencies(null)` for many <whatevers>, with a project dependency being one of them